module/zfs/zfs_ioctl.c

```diff
@@ -1683,7 +1684,12 @@ zfs_ioc_pool_scan(zfs_cmd_t *zc)
 	if ((error = spa_open(zc->zc_name, &spa, FTAG)) != 0)
 		return (error);
 
-	if (zc->zc_cookie == POOL_SCAN_NONE)
+	ASSERT(zc->zc_flags >= POOL_SCRUB_NORMAL &&
```

Comment:
Can we be a little more tolerant here and `return (SET_ERROR(EINVAL))` instead of ASSERTing `zc_flags` value?

```
root@debian-8-zfs:~# stap -g -v -d `which zfs` --ldd \
>    -e 'probe process("/lib*/libzfs.so*").function("zpool_scan").call { $cmd = 0xff; }' \
>    -c "zpool scrub -p $POOLNAME"
Pass 1: parsed user script and 113 library scripts using 94688virt/39056res/5008shr/34604data kb, in 390usr/100sys/1727real ms.
Pass 2: analyzed script: 1 probe, 1 function, 0 embeds, 0 globals using 95856virt/41876res/6536shr/35772data kb, in 20usr/0sys/118real ms.
Pass 3: using cached /root/.systemtap/cache/dd/stap_dd9957a3d2f294127e57b1abbc2c4bac_2280.c
Pass 4: using cached /root/.systemtap/cache/dd/stap_dd9957a3d2f294127e57b1abbc2c4bac_2280.ko
Pass 5: starting run.
[   72.776788] Kprobes globally unoptimized
[   72.929149] VERIFY(zc->zc_flags >= POOL_SCRUB_NORMAL && zc->zc_flags < POOL_SCRUB_FLAGS_END) failed
[   72.930278] PANIC at zfs_ioctl.c:1688:zfs_ioc_pool_scan()
[   72.932426] Showing stack for process 2024
[   72.935176] CPU: 7 PID: 2024 Comm: zpool Tainted: P           OE   4.6.4 #3
[   72.935758] Hardware name: Bochs Bochs, BIOS Bochs 01/01/2011
[   72.936396]  0000000000000286 0000000047860cbc ffff8800758a7c20 ffffffff81491063
[   72.938873]  ffffffffc0450268Message from sy 0000000000000698 ffff8800758a7c30 ffffffffc005bd04
[   72.942586]  ffff8800758a7db8 ffffffffc005bdcf 0000000000000007 ffff880000000028
[   72.943316] Call Trace:
[   72.943551]  [<ffffffff81491063>] dump_stack+0x63/0x90
[   72.943998]  [<ffffffffc005bd04>] spl_dumpstack+0x44/0x50 [spl]
[   72.944501]  [<ffffffffc005bdcf>] spl_panic+0xbf/0xf0 [spl]
[   72.946604]  [<ffffffff81200200>] ? create_object+0x170/0x2f0
[   72.948138]  [<ffffffff81812452>] ? mutex_lock+0x12/0x30
[   72.950116]  [<ffffffffc02e6433>] ? refcount_add_many+0x73/0xf0 [zfs]
[   72.950847]  [<ffffffffc02e64c6>] ? zfs_refcount_add+0x16/0x20 [zfs]
[   72.951437]  [<ffffffffc030063b>] ? spa_open_ref+0x3b/0x90 [zfs]
[   72.952031]  [<ffffffffc02f7b48>] ? spa_open_common+0x2b8/0x4b0 [zfs]
[   72.954031]  [<ffffffffc033f916>] zfs_ioc_pool_scan+0xc6/0xe0 [zfs]
[   72.956447]  [<ffffffffc0346525>] zfsdev_ioctl+0x635/0x750 [zfs]
[   72.958340]  [<ffffffff8117a934>] ? __create_xol_area+0x1a4/0x210
[   72.960631]  [<ffffffff810636fe>] ? arch_uprobe_post_xol+0x6e/0x100
[   72.962671]  [<ffffffff8121ac4d>] do_vfs_ioctl+0x9d/0x5b0
[   72.964428]  [<ffffffffc01e6617>] ? utrace_report_syscall_entry+0x117/0x130 [stap_dd9957a3d2f294127e57b1abbc2c4bac_2021]
[   72.968223]  [<ffffffff8121b1d9>] SyS_ioctl+0x79/0x90
[   72.969048]  [<ffffffff81003cf9>] do_syscall_64+0x89/0x180
[   72.970008]  [<ffffffff81814425>] entry_SYSCALL64_slow_path+0x25/0x25
```